networking/hub-regionA.json

```diff
@@ -387,9 +387,464 @@
+                                        "targetFqdns": [
+                                            "github.com",
+                                            "api.github.com"
+                                        ],
+                                        "targetUrls": [],
+                                        "terminateTLS": false,
+                                        "sourceAddresses": [],
+                                        "destinationAddresses": [],
+                                        "sourceIpGroups": [
+                                            "[resourceId('Microsoft.Network/ipGroups', variables('aksIpGroupName'))]"
+                                        ]
+                                    },
+                                    {
+                                        "ruleType": "ApplicationRule",
+                                        "name": "pull-flux-images",
+                                        "protocols": [
+                                            {
+                                                "protocolType": "Https",
+                                                "port": 443
+                                            }
+                                        ],
+                                        "fqdnTags": [],
+                                        "webCategories": [],
+                                        "targetFqdns": [
+                                            "*.docker.com",
+                                            "*.docker.io",
+                                            "docker.io",
+                                            "ghcr.io",
+                                            "github-production-container-registry.s3.amazonaws.com"
+                                        ],
+                                        "targetUrls": [],
+                                        "terminateTLS": false,
+                                        "sourceAddresses": [],
+                                        "destinationAddresses": [],
+                                        "sourceIpGroups": [
+                                            "[resourceId('Microsoft.Network/ipGroups', variables('aksIpGroupName'))]"
+                                        ]
+                                    }
+                                ],
+                                "name": "Flux-Requirements",
+                                "priority": 300
+                            }
+                        ]
+                    }
+                },
+                {
+                    "type": "Microsoft.Network/firewallPolicies/ruleCollectionGroups",
+                    "apiVersion": "2020-11-01",
+                    "name": "[concat(variables('hubFwName'), '/DefaultNetworkRuleCollectionGroup')]",
+                    "location": "[parameters('location')]",
+                    "dependsOn": [
+                        "[variables('hubFwName')]",
```

Comment:
Doesn't this depend on the ipGroup?  Looks like ipGroup dependencies are not here?  Same for the others in here.

Reply:
the dependency is on the root object, then the children are not needed to be added. 
You need to add on the children, if you define the ruleCollectionGroups outside the Microsoft.Network/firewallPolicies scope.

 "type": "Microsoft.Network/firewallPolicies",
            "apiVersion": "2020-11-01",
            "name": "[variables('hubFwName')]",
            "location": "[parameters('location')]",
            "dependsOn": [
                **"[resourceId('Microsoft.Network/ipGroups', variables('aksIpGroupName'))]"**
            ],

Reply:
That seems like that could cause a issue if someone breaks the dependency chain here, especially since the dependency chain is currently kinda a workaround and artificial.  Would it be wrong to add that here?

Reply:
Would it be wrong to add that to each one?
no, but I preferred  avoid redundancy. 
I could change if you prefer.

The artificial dependency among  ruleCollectionGroups is not playing here. 
Here the dependency playing is between the parent resource (Microsoft.Network/firewallPolicies) and its children resources (the collection of Microsoft.Network/firewallPolicies/ruleCollectionGroups)

Reply:
You're right about that, I should have looked closer when I commented, but that resource doesn't ACTUALLY depend on the ip group, so I think it would be more correct to only list it (even if duplicated) on the ones that to.

Reply:
ok, I will do it.

Reply:
it was moved down